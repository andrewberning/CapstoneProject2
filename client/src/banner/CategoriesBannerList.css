.CategoriesBannerList {
  margin: 50px;
  max-width: 1000px;
}

.CategoriesBannerList .categories-list {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-evenly;
  padding: 0;
  margin: 0;
}

.CategoriesBannerList .categories-list li {
  flex: 1 1 calc(25% - 20px); 
  max-width: calc(25% - 20px);
}

@media (max-width: 768px) {
  .CategoriesBannerList .categories-list li {
    flex: 1 1 calc(45% - 20px); 
    max-width: calc(45% - 20px);
    padding-bottom: 20px;
  }
}

@media (max-width: 480px) {
  .CategoriesBannerList .categories-list li {
    flex: 1 1 100%; 
    max-width: 100%;
  }
}
