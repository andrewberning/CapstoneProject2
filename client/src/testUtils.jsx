import React from "react";
import UserContext from "./auth/UserContext";
import { vi } from 'vitest';

const demoUser = {
  username: "testuser",
  first_name: "Test",
  last_name: "User",
  email: "testuser123@test.net",
};

// Mock functions
const mockSetCurrentUser = vi.fn();
const mockLogin = vi.fn();
const mockLogout = vi.fn();
const mockSignup = vi.fn();
const mockAddToCart = vi.fn();
const mockHasAddedToCart = vi.fn();

const UserProvider = ({ 
  children,
  currentUser = demoUser,
  setCurrentUser = mockSetCurrentUser,
  login = mockLogin,
  logout = mockLogout,
  signup = mockSignup,
  addToCart = mockAddToCart,
  hasAddedToCart = mockHasAddedToCart,
  cartItems = [],
  setCartItems = () => {},
  categories = [],
  infoLoaded = true,
  totalAmount = 0,
  totalItems = 0,
}) => (
  <UserContext.Provider value={{ 
    currentUser,
    setCurrentUser,
    login,
    logout,
    signup,
    addToCart,
    hasAddedToCart,
    cartItems,
    setCartItems,
    categories,
    infoLoaded,
    totalAmount,
    totalItems,
  }}>
    {children}
  </UserContext.Provider>
);

export { UserProvider, mockSetCurrentUser, mockLogin, mockLogout, mockSignup, mockAddToCart, mockHasAddedToCart };
