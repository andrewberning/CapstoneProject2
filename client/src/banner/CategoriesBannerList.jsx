import { useContext } from "react";
import CategoryCard from "../categories/CategoryCard";
import UserContext from "../auth/UserContext";
import "./CategoriesBannerList.css";

export default function CategoriesCardList() {
  const { categories } = useContext(UserContext);
  
  return (
    <div className="CategoriesBannerList">
      <ul className="categories-list">
      {categories.map(category => (
        <CategoryCard
          key={category.id}
          id={category.id}
          category={category.name}
          image={category.image_url}
        />
      )).slice(0,4)}
      </ul>
    </div>
  );
}