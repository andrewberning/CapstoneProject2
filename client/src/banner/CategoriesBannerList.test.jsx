import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CategoriesBannerList from "./CategoriesBannerList"
import { UserProvider } from "../testUtils";

describe('CategoriesBanner component', () => {
  it('renders without crashing', () => {
    render(
      <MemoryRouter>
        <UserProvider >
          <CategoriesBannerList />
        </UserProvider>
      </MemoryRouter>
    );
  });

  it("matches snapshot", function () {
    const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <CategoriesBannerList />
        </UserProvider>
      </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});