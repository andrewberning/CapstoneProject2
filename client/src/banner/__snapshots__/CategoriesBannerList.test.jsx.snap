// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoriesBanner component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CategoriesBannerList"
  >
    <ul
      class="categories-list"
    />
  </div>
</DocumentFragment>
`;
